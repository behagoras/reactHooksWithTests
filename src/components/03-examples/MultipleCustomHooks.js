import React, { useEffect, useState } from 'react'
import useCounter from '../../hooks/useCounter'
import useFetch from '../../hooks/useFetch'

export default function MultipleCustomHooks() {
  const baseUrl = 'https://www.breakingbadapi.com/api'

  const [
    counter,
    increment,
    decrement,
    reset,
  ] = useCounter(1, 1)
  const [quoteUrl, setQuoteUrl] = useState(`${baseUrl}/quotes/${counter}`)
  const state = useFetch(quoteUrl)

  const { data = [], error, loading } = state

  useEffect(() => { setQuoteUrl(`${baseUrl}/quotes/${counter}`) }, [counter])

  return (
    <div>
      <h1>Breaking Bad characters</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">
          Loading...
        </div>
      ) :
        data && data.map((quote, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <blockquote className="blockquote text-right" key={index}>
            <p className="mb-0">{quote?.quote}</p>
            <footer className="blockquote-footer">{quote?.author}</footer>
          </blockquote>
        ))}
      {/* {data && <pre>{JSON.stringify(data, 0, 1)}</pre>} */}
      <div className="buttons text-right">
        <button
          disabled={counter <= 1}
          type="button"
          className="btn btn-primary "
          onClick={counter > 1 && decrement}
        >
          {'<'}
        </button>
        <button
          type="button"
          className="btn "
          onClick={reset}
        >
          Reset
        </button>
        <button
          type="button"
          className="btn btn-primary"
          onClick={increment}
        >
          {'>'}
        </button>
      </div>
      {error && <pre>{JSON.stringify(error, 0, 1)}</pre>}
    </div>
  )
}

