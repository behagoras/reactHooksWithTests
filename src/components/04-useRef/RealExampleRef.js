import React from 'react'

export default function RealExampleRef() {
  return (
    <div>
      <h1>Real Example Ref</h1>
      <hr />
    </div>
  )
}
