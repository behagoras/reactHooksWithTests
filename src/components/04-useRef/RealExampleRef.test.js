import { shallow } from 'enzyme'
import React from 'react'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'
import RealExampleRef from './RealExampleRef'

describe('<RealExampleRef />', () => {
  it('should render correctly', () => {
    const component = shallow(<RealExampleRef />)
    expect(component).toMatchSnapshot()
    expect(component.find(MultipleCustomHooks).exists()).toBe(false)
  })
  it('should display the component <MultipleCustomHooks />', () => {
    const component = shallow(<RealExampleRef />)
    component.find('button').simulate('click')
    expect(component.find(MultipleCustomHooks).exists()).toBe(true)
  })
})
