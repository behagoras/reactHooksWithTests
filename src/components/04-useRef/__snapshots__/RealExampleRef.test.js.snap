// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> should render correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
    type="button"
  >
    Show/Hide
  </button>
</div>
`;
